### ISSUE TITLE
System.DllNotFoundException: Unable to load DLL 'e_sqlite3' with Xamarin .NET Core UWP 

### ISSUE THREAD

**Issue by alaskanrogue** [ericsink/SQLitePCL.raw#322]
In the UWP project of a Xamarin .NET Core cross-platform, encountering the following exception; the other projects do not have the issue:

> System.IO.FileNotFoundException
  HResult=0x80070002
  Message=Could not load file or assembly 'e_sqlite3, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Reflection.RuntimeAssembly.nLoad(AssemblyName fileName, String codeBase, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, IntPtr ptrLoadContextBinder)
   at System.Reflection.RuntimeAssembly.InternalLoadAssemblyName(AssemblyName assemblyRef, RuntimeAssembly reqAssembly, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, IntPtr ptrLoadContextBinder) in E:\A\_work\658\s\src\mscorlib\src\System\Reflection\RuntimeAssembly.cs:line 353

[Output-Debug.txt](https://github.com/ericsink/SQLitePCL.raw/files/4011655/Output-Debug.txt)
![Call Stack](https://user-images.githubusercontent.com/14001196/71606710-6ffde400-2b41-11ea-9c5a-72ff0fc27c0a.png)

**Comment by ericsink** [comment]
So far I';m having trouble making sense of this.  The stack trace doesn't seem to be related to SQLite, and the exception mentioning e_sqlite3 seems to be coming out of nowhere.

FWIW, my UWP tests seem to work just fine, and a number of other folks are having no problems with UWP.

Perhaps if you provide some more info we will find a clue.

What version of SQLitePCLRaw?

Did this previously work and then it stopped?  If so, what changed?

What are all the nuget packages are you using?

Can you provide a small project to reproduce the problem?

**Comment by alaskanrogue** [comment]
It appears to be a Xamarin Forms Init error. With a days work I might be able to create a small project. Additional details attached.

[Nugets.txt](https://github.com/ericsink/SQLitePCL.raw/files/4012528/Nugets.txt)
![Call Stack](https://user-images.githubusercontent.com/14001196/71619907-c21c2500-2b94-11ea-99bf-25fedeea5444.png)
![StackMap](https://user-images.githubusercontent.com/14001196/71619908-c6484280-2b94-11ea-95c9-a8c82d87e850.png)
![ThreadView](https://user-images.githubusercontent.com/14001196/71619913-c9dbc980-2b94-11ea-92ea-42c5b6ed72ee.png)
[RelevantStackTrace.txt](https://github.com/ericsink/SQLitePCL.raw/files/4012533/RelevantStackTrace.txt)
[Output-Debug.txt](https://github.com/ericsink/SQLitePCL.raw/files/4012535/Output-Debug.txt)

**Comment by alaskanrogue** [comment]
`        /// <summary>
        /// Invoked when the application is launched normally by the end user
        /// </summary>
        protected override async void OnLaunched(LaunchActivatedEventArgs e)
        {
            Frame rootFrame = Window.Current.Content as Frame;

            // Do not repeat app initialization when the Window already has content,
            // just ensure that the window is active
            if (rootFrame == null)
            {
                // Create a Frame to act as the navigation context and navigate to the first page
                rootFrame = new Frame();

                rootFrame.NavigationFailed += OnNavigationFailed;

                IEnumerable<System.Reflection.Assembly> a = null;

                Xamarin.Forms.Forms.Init(e, a);

                if (e.PreviousExecutionState == ApplicationExecutionState.Terminated)
                {
                    //TODO: Load state from previously suspended application
                }

                // Place the frame in the current Window
                Window.Current.Content = rootFrame;
            }
`

**Comment by alaskanrogue** [comment]
Both e_sqlite3 and SQLitePCLRaw.provider.e_sqlite3 are version 1.0 dated 1/4/17 and present in the UWP\bin folders.

**Comment by ericsink** [comment]
So it looks like your only use of SQLitePCLRaw is transitively coming from AppCenter.  Does that sound right?

**Comment by alaskanrogue** [comment]
That sounds possibly correct as with the other Nugets I don't see as needing databases. As the OnLaunch is prior to the actual PCL initialization, it shouldn't have anything to do with the PCL Nugets. Demo with issue attached.

[SqlLite.zip](https://github.com/ericsink/SQLitePCL.raw/files/4013301/SqlLite.zip)

**Comment by ericsink** [comment]
When I run that demo, I get a null reference exception that seems unrelated to the issue here.  Do I need to do anything special?

**Comment by alaskanrogue** [comment]
When I initially encountered the issue, that was the exception I searched and eventually asked about on the MS Dev. Community. 

> **Xamarin.Forms.Application.set_MainPage(Page value) is passed a NULL Page Value**: During startup of a Xamarin cross-platform UWP app, a !DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION is thrown stating that Xamarin.Forms.Application.set_MainPage(Page value) is passed a NULL value.

> This began after a recent upgrade of numerous NUGETs. Debugging in native mode shows that a value is present in the code line before the function is executed. The same error is presented in an imported templated UWP application in the same solution.

As I dug deeper, I reported: 

>**Window.Current.Activate Exception**: In a cross-platform Xamarin UWP app, Window.Current.Activate is encountering an exception which appears to be a {“Object reference not set to an instance of an object.”}.

Having fully turned on debugging, I.e. verbose output, symbol loading, stepping through the libraries, I got to were I am today.

**Comment by ChrisProlls** [comment]
The same error occurs in our side with a ASP.NET Core project and the EF Core library.

**Comment by alaskanrogue** [comment]
Last week I solved the problem by returning to a copy of the last working version, copying and pasting the .cs and XAML files with the other relevant configuration files into that solution. I then compiled to determine if the problem was present. It wasn't. Then I slowly upgraded the nuggets through the versions, initially targeting Xamarin.Forms Nugget. I built each, testing to see if the problem appeared. I successfully reached the current version. I then did the same to the other Nuggets. (Luckily I save independent version copies of the solution as I develop.)
